File: LeetCode/796/Solution.cpp
class Solution {
public:    
    bool rotateString(string A, string B) {
        if(A.length() != B.length())
            return false;
        else if(A == "" && B == "")
            return true;
        
        int n = A.length();
        
        for(int i=0;i<n;++i) {
            bool flag = true;

            for(int j=0;j<n;++j) {
                if(A[j] != B[(i + j) % n]) {
                    flag = false;
                    break;
                }
            }
            
            if(flag)
                return true;
        }
        
        return false;
    }
};

File: LeetCode/796/Solution.java
class Solution {
    public boolean rotateString(String s, String goal) {
        if(s.length() != goal.length()) {
            return false;
        }

        for(int i=0;i<s.length();++i) {
            int count = 0;
            for(int j=0;j<goal.length();++j) {
                if(s.charAt(j) != goal.charAt((i+j)%goal.length())) {
                    break;                    
                }
                count++;
            }
            if(count == goal.length()) {
                return true;
            }
        }

        return false;
    }
}
